### ISSUE TITLE
feat: treesitter semantic context

### ISSUE THREAD

**Issue by rorybyrne** [flywhl/anot#2]

**Comment by oyarsa** [comment]
I'd like to this one, if there are no objections. Any restrictions on what this could look like?

**Comment by rorybyrne** [comment]
Yes please. No strong opinions about what it looks like, the structure can be changed before v1 anyway.

Looking at the PR now, will likely merge tonight. Thank you!
